feather/plugin-host/src/host_calls/plugin_message.rs

```diff
@@ -0,0 +1,24 @@
+use feather_common::events::PluginMessageEvent;
+use feather_plugin_host_macros::host_function;
+
+use crate::context::{PluginContext, PluginPtr};
+
+#[host_function]
+pub fn plugin_message_send(
+    cx: &PluginContext,
+    channel_ptr: PluginPtr<u8>,
+    channel_len: u32,
+    data_ptr: PluginPtr<u8>,
+    data_len: u32,
+) -> anyhow::Result<()> {
+    let channel = cx.read_string(channel_ptr, channel_len)?;
+    let data = cx.read_bytes(data_ptr, data_len)?;
+
+    let event = PluginMessageEvent {
+        channel,
+        data,
+    };
+    cx.game_mut().ecs.insert_event(event);
```

Comment:
Plugin messages should be associated with a single entity/player for the query on `ClientId` to work. We should add a new entity parameter (`u64`) to the host call that indicates the player to send the plugin message to. Then use `insert_entity_event(Entity::from_bits(entity_u64), event)`.